### DISCUSSION TITLE
Developement discussion

### DISCUSSION THREAD

**Discussion by MadOne** [OStrama/weishaupt_modbus#12] [General]
Hi,

i have a few questions/suggestions for the futher developement.

- I suggest to "unfork" this repo. It has nearly nothing in common with the old one. Also remove the Info in readme https://github.com/orgs/community/discussions/52663
- We should try to catch the users of my old integration. I am not sure about the best approach. Is there a possibility of forwarding to this repo? Or just do a release with the information (I think most users wont read). Or maybe intentionally break the integration to wake up the users to switch.
- I activated the necessary checks for HACS: Error: R] [MANIFEST] Domain does not match dir name . So we have to either rename the domain or change the repo name. Both will break instrallations :(. This is also relevant for the icons, as they are tied to the domain.

**Comment by OStrama** [comment]
Yes I also thought about this. We should have a look at the "install base": Move to the naming that the majority of users use. I have no idea how many and how active they are.

The most annoying would be a change of the entity IDs, especially when using them for long time statistics, i.e. energy dashboards. In the newest HA version there was an anouncement that statistics can be attached to entities easier than before. Maybe, I do an upgrade to check.

Yes, since I use the remote environment including reformatting, I also remove warnings etc. when editing a file.

**Comment by MadOne** [comment]
There are a lot possibilities but none of them are flawless.

1. We could change back the DOMAIN to weishaupt_modbus: This would fix the icon and stop the HACS checks from complaining. But will break the entitys as the unique id is dependent on the domain for now.
self._attr_unique_id = CONST.DOMAIN + self._attr_name ->self._attr_unique_id = CONST.DOMAIN_OLD + self._attr_name is a hacky fix.
Users would have to setup a new configuration for the heatpump. I think statistics values would be found if we take the workaround above.
This might also be a good time to introduce the prefix as the prefix would break the unique_id and also the statistics anyway.

2. Change the repository name and the folder name: This will disconnect HACS users from the integration developement. They would have to add a new repo to HACS.

I like none of the above. It would be sad if we loose some users, On the other side we might just ignore the consequences and do it right. But then we sould introduce all breaking changes at once.
For now my prefered solution is variant one with breaking the statistics and introduce the prefix.

**Comment by OStrama** [comment]
I‘d also prefer 1.
When we implement the prefix before, there‘s a simple migration path:
- standard prefix will be Weishaupt Modbus
- Users who used weishaupt wbb before have the choice:
      a) change prefix to it and keep statistics
      b) use new default and re-assign statistics via database hack (see Simon42 lost entity video)
      c) use new default and start with new statistics

we only have to give clear instructions in the readme and perhaps in the config dialog..

**Comment by MadOne** [comment]
I changed the domain to weishaupt_modbus.
The Prefix comes from const.py for now and is set to weishaupt_wbb.
You have to remove the heatpump and setup a new one, because of the changed domain.
This fixes the logos and HACS / hassfest checks.

There is a possibility in the config flow to react to updates. So we could devide in users update from 0.0.xx to  > 0.0.8 and users who start with a fresh install > 0.0.8

**Comment by OStrama** [comment]
Good Idea! I had a look at other custom integrations (Fronius). I think I understood how to add custom parameters. When ready we can put the prefix in as a parameter.

**Reply by MadOne** [reply]
I changed modbusclient.disconnect() to sync. This is acording to the pymodbus example. I guess the disconnect is only client sided so no async is needed. This threw an error the whole time, but we didnt see it because if the bare try except. Now nearly all values are up. 

There is still a lot of debugging to do. We have to get rid of those  bare try except statements.

**Reply by OStrama** [reply]
Thanks, I took it over and enabled the heat power again. I'll try to do a pre-release that I can test it on my production system for a day or two.

My problem with the try-except sectios is, that I did not find a documentation of possible exceptions. (I know it from C++ docu that a function description usually contains infos like "throws the following exceptions:")

**Reply by MadOne** [reply]
I think you can select a branch in hacs. So no need for a release.

What exceptions are you missing? The modbus exception is displayed when i hover over the functions.

**Comment by OStrama** [comment]
Can you have a look at the readme? I tried to put in some migration info...

**Reply by MadOne** [reply]
Im in work right now. I hope i have time to think about the Migration when im at home. Too difficult on the phone

**Comment by OStrama** [comment]
Some insights from my production test.  the number of faults decreased compared with my tests in the devenv. Maybe, because now we do not have two instances talking with the heatpump. 

Nevertheless, 

- I have stable entity values with no gaps anymore
- I can control the parameters reliably

I see the following in the log:

![{7E268D9D-292A-4C66-A9D9-C5DE2DE656C2}](https://github.com/user-attachments/assets/bd4d0ade-e8f2-4cbb-b81e-63077a9b7fc9)

the 1st one happens constantly:

![{CFA70FC0-97D0-449A-AE8B-7D627D6B016A}](https://github.com/user-attachments/assets/615d87b2-377e-4832-bcaa-45f7857bf7df)

Connection failures from time to time:

![{A516AA0A-0ABD-4431-BB9D-85CC35637294}](https://github.com/user-attachments/assets/0d5cfa00-2436-46da-9125-203f10293680)

![{B366F702-D8BE-460B-87D9-0E67D477F293}](https://github.com/user-attachments/assets/a4db4063-3f33-43c9-aedc-e2bf72e3bfa1)

But these only at startup:

![{6D84E79A-E0C5-4CE3-A651-F0BBE6F43150}](https://github.com/user-attachments/assets/d1e89068-6e2e-4f06-96f9-bf7334b85f31)

**Comment by OStrama** [comment]
Digging a bit deeper it looks more clear:

At the beginning every request fails. You see it also in slow updating of entities after 1st start-up:

	Zeile  13: 2024-10-25 11:25:44.540 WARNING (MainThread) [py.warnings] /home/homeassistant/.homeassistant/custom_components/weishaupt_modbus/modbusobject.py:52: UserWarning: Connection to heatpump succeeded
	Zeile  16: 2024-10-25 11:25:44.541 WARNING (MainThread) [py.warnings] /home/homeassistant/.homeassistant/custom_components/weishaupt_modbus/entities.py:201: UserWarning: Item:Aussentemperatur failed

	.
.
.

	Zeile 268: 2024-10-25 11:25:44.545 WARNING (MainThread) [py.warnings] /home/homeassistant/.homeassistant/custom_components/weishaupt_modbus/entities.py:201: UserWarning: Item:Elektr. Energie Monat failed
	Zeile 271: 2024-10-25 11:25:44.545 WARNING (MainThread) [py.warnings] /home/homeassistant/.homeassistant/custom_components/weishaupt_modbus/entities.py:201: UserWarning: Item:Elektr. Energie Jahr failed

In normal operation, the two others only happen mostly at exactly the same time. Maybe because of asynch HA tries 2 requests in parallel?

	Zeile 274: 2024-10-25 11:26:14.690 WARNING (MainThread) [py.warnings] /home/homeassistant/.homeassistant/custom_components/weishaupt_modbus/modbusobject.py:52: UserWarning: Connection to heatpump succeeded
	Zeile 278: 2024-10-25 11:26:14.712 WARNING (MainThread) [py.warnings] /home/homeassistant/.homeassistant/custom_components/weishaupt_modbus/entities.py:227: UserWarning: connection to the heatpump failed
	Zeile 356: 2024-10-25 11:48:14.691 WARNING (MainThread) [py.warnings] /home/homeassistant/.homeassistant/custom_components/weishaupt_modbus/modbusobject.py:52: UserWarning: Connection to heatpump succeeded
	Zeile 359: 2024-10-25 11:48:14.691 WARNING (MainThread) [py.warnings] /home/homeassistant/.homeassistant/custom_components/weishaupt_modbus/entities.py:227: UserWarning: connection to the heatpump failed
	Zeile 601: 2024-10-25 13:05:14.690 WARNING (MainThread) [py.warnings] /home/homeassistant/.homeassistant/custom_components/weishaupt_modbus/modbusobject.py:52: UserWarning: Connection to heatpump succeeded
	Zeile 604: 2024-10-25 13:05:14.690 WARNING (MainThread) [py.warnings] /home/homeassistant/.homeassistant/custom_components/weishaupt_modbus/entities.py:227: UserWarning: connection to the heatpump failed
	Zeile 746: 2024-10-25 13:47:44.690 WARNING (MainThread) [py.warnings] /home/homeassistant/.homeassistant/custom_components/weishaupt_modbus/modbusobject.py:52: UserWarning: Connection to heatpump succeeded
	Zeile 749: 2024-10-25 13:47:44.690 WARNING (MainThread) [py.warnings] /home/homeassistant/.homeassistant/custom_components/weishaupt_modbus/entities.py:227: UserWarning: connection to the heatpump failed

**Comment by MadOne** [comment]
I managed to reduce the errors  and also get a "IllegalAddress" response. This could be used to filter the modbus addresses not supported by the heatpump.

**Comment by OStrama** [comment]
I installed it some minutes ago. I only see this

2024-10-25 21:09:14.395 WARNING (MainThread) [py.warnings] /home/homeassistant/.homeassistant/custom_components/weishaupt_modbus/modbusobject.py:52: UserWarning: Connection to heatpump succeeded
warnings.warn("Connection to heatpump succeeded")
2024-10-25 21:09:14.939 ERROR (MainThread) [pymodbus.logging] Cancel send, because not connected!

I have some not working entities, too. But I do not see messages from them..

**Reply by MadOne** [reply]
Run it in dev Container and you will see. 
I also read about an integration that can display debug info.

**Reply by OStrama** [reply]
Thanks! Will do tomorrow..

**Comment by OStrama** [comment]
Removed another await on close which crashed entity changes ;-) Made an update

**Comment by MadOne** [comment]
I pushed out a new release because i made some heavy progress.

**Comment by OStrama** [comment]
Great! I used the new flag for checking is an item should be added to the list. It works for my WEZ2.
Have to leave now. When I have some time tomorrow, I'll add all entities to the list that are in the Excel Sheet..

**Comment by OStrama** [comment]
When looking at the Excel sheet with the addresses: A temperature value of -32k means not available. Should we treat them as _is_invalid, too?

**Reply by MadOne** [reply]
I think so. It unlikely hat the appear mid session.

**Comment by OStrama** [comment]
Another point: It seems that the connections sometimes left open and that there‘s an issue with too many open files after a while. Had to restart my Pi after HA was not responding anymore. That never happens before 0.0.8..

**Reply by MadOne** [reply]
You seem to have fixed this perfectly fine. I really like this.

**Comment by OStrama** [comment]
I've seen that you removed the reconnect at ModbusObject. Doesn't it prevent reading data if the heatpump is offline for a while and gets online again?

Maybe we should use the autoconnect/autoclose feature? https://pymodbustcp.readthedocs.io/en/latest/package/class_ModbusClient.html

**Reply by MadOne** [reply]
I think i it stays "connected" forever untill it gets actively closed(), what we only do on integration unload.
But we should definitly try that out. If problems occour the autoconnect looks good for me.

I wouldnt use the autoclose, because it would close after each request so we would spam connect/disconnect again.

**Comment by MadOne** [comment]
Do you have discord?
I opened a group for easier communication:
https://discord.gg/gXE55MhH

**Comment by OStrama** [comment]
OK, then I set autoconnect true and give it a try.
BTW: Tried to add check_valid vor temperatures in Modbusobject, but it seems to fail, since the 1st call always returns None.
And Heatpower fails, since some temperatures are in Range 1,5000

**Reply by MadOne** [reply]
> BTW: Tried to add check_valid vor temperatures in Modbusobject, but it seems to fail, since the 1st call always returns None.
I am not sure if i understand you right. mbo._is_invalid is only set after it has checked a value. So we have to check a value before checking if it is valid.

**Reply by MadOne** [reply]
look [this](https://github.com/OStrama/weishaupt_modbus/commit/7983cce2998e1bcfe2b9f9759f6ec7a4301bbb6b)

**Comment by OStrama** [comment]
Made a pre5, just to check on production system and to get feedback from users..

**Comment by OStrama** [comment]
Im about updating config flow..

**Reply by MadOne** [reply]
Nice. Didnt wrap my head around that yet.

I wrote a [small script](https://github.com/OStrama/weishaupt_modbus/blob/main/custom_components/weishaupt_modbus/test_modbus.py) that checks all possoble modbus addresses. Would be nice if you can run that too.

**Comment by MadOne** [comment]
I have pulled all responding addresses from the hp and compared these to the doku / our values.
This addresses are neither documented nor in our integration.
33108 & 33109 seem not installed
44105 & 44106 look like a error to me too.

But 33107 & 33111 are interesting.

33107 | 115
33108 | 32768
33109 | 32768
33110 | 0
33111 | 425
44105 | 65486
44106 | 65486

**Reply by OStrama** [reply]
Tried to run the script in the devcontainer, get error message: 

(ha-venv) vscode ➜ ~/weishaupt_modbus (main) $ /home/vscode/.local/ha-venv/bin/python /home/vscode/weishaupt_modbus/custom_components/weishaupt_modbus/test_modbus.py
Traceback (most recent call last):
  File "/home/vscode/weishaupt_modbus/custom_components/weishaupt_modbus/test_modbus.py", line 1, in <module>
    import asyncio
  File "/usr/local/lib/python3.12/asyncio/__init__.py", line 8, in <module>
    from .base_events import *
  File "/usr/local/lib/python3.12/asyncio/base_events.py", line 24, in <module>
    import socket
  File "/usr/local/lib/python3.12/socket.py", line 55, in <module>
    import os, sys, io, selectors
  File "/usr/local/lib/python3.12/selectors.py", line 12, in <module>
    import select
  File "/home/vscode/weishaupt_modbus/custom_components/weishaupt_modbus/select.py", line 3, in <module>
    from homeassistant.config_entries import ConfigEntry
  File "/workspaces/core/homeassistant/config_entries.py", line 31, in <module>
    from . import data_entry_flow, loader
  File "/workspaces/core/homeassistant/data_entry_flow.py", line 21, in <module>
    from .core import HomeAssistant, callback
  File "/workspaces/core/homeassistant/core.py", line 100, in <module>
    from .util import dt as dt_util, location
  File "/workspaces/core/homeassistant/util/location.py", line 12, in <module>
    import aiohttp
  File "/home/vscode/.local/ha-venv/lib/python3.12/site-packages/aiohttp/__init__.py", line 6, in <module>
    from .client import (
  File "/home/vscode/.local/ha-venv/lib/python3.12/site-packages/aiohttp/client.py", line 39, in <module>
    from . import hdrs, http, payload
  File "/home/vscode/.local/ha-venv/lib/python3.12/site-packages/aiohttp/http.py", line 7, in <module>
    from .http_parser import (
  File "/home/vscode/.local/ha-venv/lib/python3.12/site-packages/aiohttp/http_parser.py", line 28, in <module>
    from .base_protocol import BaseProtocol
  File "/home/vscode/.local/ha-venv/lib/python3.12/site-packages/aiohttp/base_protocol.py", line 4, in <module>
    from .client_exceptions import ClientConnectionResetError
  File "/home/vscode/.local/ha-venv/lib/python3.12/site-packages/aiohttp/client_exceptions.py", line 264, in <module>
    class ServerTimeoutError(ServerConnectionError, asyncio.TimeoutError):
                                                    ^^^^^^^^^^^^^^^^^^^^
AttributeError: partially initialized module 'asyncio' has no attribute 'TimeoutError' (most likely due to a circular import)

What do I do wrong?

**Comment by MadOne** [comment]
Nothing. Does not work in dev container.
Run it locally.

**Comment by MadOne** [comment]
Im missing a LOT of values in the new pre release.

**Reply by MadOne** [reply]
Only 3 devices. No numbers no sensors. Only some selects.

**Reply by MadOne** [reply]
Pre 4 looks like the last looking version

**Reply by OStrama** [reply]
Works on my side? This seems to be an issue with check of valid sensors?

**Reply by OStrama** [reply]
Are they completely missing or Unknown? If unknown: Did you wait for a while, sometimes it takes 1-2 cycles to see all sensors.

**Reply by MadOne** [reply]
Might be a rest of the config from my integration. Hacs complaned about not removing the config at uninstall.  I will try to cleanup that mess tomorrow and do a fresh Install

**Reply by OStrama** [reply]
Ah yes, that's possible. On my side it works..

**Reply by MadOne** [reply]
But that could mean everyone with my previous addon has a potential problem.  I have to fix that

**Reply by OStrama** [reply]
Yes, that's true. Did you change the prefix in the config dialog?

**Reply by OStrama** [reply]
I made another check from scratch and saw the following behavior:

- after installation, only 3 devices were shown in the "choose location" screen. After updating the 1st location, all devices appeared.
- when opening the 1st device, no entry was shown. After pressing F5 they all got available.

maybe, a browser cache issue?

**Reply by MadOne** [reply]
I dont know, but this behaviour sounds like the one i faced.

**Reply by OStrama** [reply]
OK, I agree with you: In general it became unstable after adding all the mechanics to not add entities that are not showing useful data, like temperature -32768 etc. 

What do you think about following:
- We only do not add entities that generate an exception. 
- Entities that have values, even if they are showing "no sensor connected" stay in
- I add a parameter about number of Heizkreise to add

So we can go back to the pre4 mechanics that may be most stable.

I think thiis all only plays a role at installation, which doesn't happen so often...

**Reply by MadOne** [reply]
I really like the auto detect, but maybe this should be a feature for future versions. We should really bring out a new stable version.

**Reply by MadOne** [reply]
Another idea:
Add all values and maniplate them on update
https://developers.home-assistant.io/docs/core/entity for example available. Not sure what the correct options are here and what works.

**Reply by MadOne** [reply]
Even without the filtering this happens.

Maybe because of the increased entity number with HK2?

**Reply by OStrama** [reply]
Possible.. What happens if you comment them out?

**Reply by MadOne** [reply]
Still the same.

**Comment by OStrama** [comment]
I make a new pre, since I found a device string error, but that will not solve your issue...

**Comment by OStrama** [comment]
This is my register-file:

[register.txt](https://github.com/user-attachments/files/17549643/register.txt)

**Reply by MadOne** [reply]
I addes your values to the ["Auswertung"](https://github.com/OStrama/weishaupt_modbus/blob/main/custom_components/weishaupt_modbus/auswertung_register.csv)

33106, 33107,33111 seem to be interesting. We should add them and monitor them. So maybe we can guess what they are.

**Reply by OStrama** [reply]
made a new pre:
- migration of config entry
- added unknown sensors in device "unknown"

**Reply by OStrama** [reply]
See issues from backuprestore for content of the found sensors..

**Reply by edlestau** [reply]
Hi there .
At my Biblock seems to be
33106  Verdampfungstemperatur
33107  Verdichtersauggastemp.
You can see this values at the portal when using "fachmann" access

**Reply by Cheetah2021** [reply]
Hi, 
In addition at my BiBlock with solar thermal system it seems to be:
33108; Weichentemperatur
33109; Vorlauf regenerativ
33110; Puffertemperatur
33111; Summenvorlauf (B7) ["Wärmepumpe" in WEM Portal]

**Reply by MadOne** [reply]
> Hi, In addition at my BiBlock with solar thermal system it seems to be: 33108; Weichentemperatur 33109; Vorlauf regenerativ 33110; Puffertemperatur 33111; Summenvorlauf (B7) ["Wärmepumpe" in WEM Portal]

I updated the descriptions with [this](https://github.com/OStrama/weishaupt_modbus/commit/faf36c2c0532c74faa78b982099333ccbdfaf312)

**Reply by OStrama** [reply]
Great! It gets more and more comprehensive:-)

**Comment by MadOne** [comment]
HA complains about file io (kennfeld.py) is not async

**Reply by OStrama** [reply]
Yes, I know. Have to implement file r/w async. (and have to learn how ;-)

**Reply by OStrama** [reply]
changed it to async..

Including all the other changes: Would it make sense to make an official release with the state we have?

**Reply by MadOne** [reply]
Yes we should create a new release. I d like to fix the device problem at Installation before, but all atempts failed.

**Reply by MadOne** [reply]
![Bildschirmfoto vom 2024-10-30 19-26-49](https://github.com/user-attachments/assets/f5d33fa6-68f8-4372-bf84-500800da27ef)
Do you see the missing captions? This is required for a realease in my opinion.

**Reply by OStrama** [reply]
For me it looks like this:
![image](https://github.com/user-attachments/assets/963f1498-c1c4-4de8-9400-82724b88b390)

**Reply by OStrama** [reply]
It takes the names of the config entries but not the strings from the files..

**Reply by OStrama** [reply]
It Works!

![image](https://github.com/user-attachments/assets/1c1ee4f2-ee09-43f2-9c62-2c21ed276a15)

- according to https://community.home-assistant.io/t/hacs-custom-integration-config-flow-strings-are-not-loading/584597/3 the translations files are not auto-generated for custom integrations. So we have to generate all language files by ourselves. I simply copied the string.json to translation and it worked..

**Reply by OStrama** [reply]
I'll check in this now in en and de. Please feel free to adapt ;-)

![image](https://github.com/user-attachments/assets/061a57b7-4de4-4462-b812-18b73797e9c3)

**Reply by MadOne** [reply]
Good job. This is a big step.

**Comment by MadOne** [comment]
What about adding our integration to HACS.
Acording to:
https://www.hacs.xyz/docs/publish/include/
we have all requirements.
Whis would enable people to find our integration much easier.

**Reply by OStrama** [reply]
Yes, that would be a good idea! (I‘m out this weekend, sorry for late replies..)

**Comment by MadOne** [comment]
When playing around with the translation i did: python -m script.hassfest   

It reported:
Integration weishaupt_modbus:
* [ERROR] [MANIFEST] Invalid manifest: extra keys not allowed @ data['version']. Got '0.0.8'
* [ERROR] [REQUIREMENTS] Requirement pymodbus need to be pinned "<pkg name>==<version>".
* [ERROR] [REQUIREMENTS] Requirement aiofiles need to be pinned "<pkg name>==<version>".

Not sure what the complain about version is, maybe because its 0.x.x?
Should we pin the requirements?

**Reply by OStrama** [reply]
What do they mean with pinning? Mention them in requirements?

**Reply by MadOne** [reply]
Pin means to specify the exact version

**Reply by OStrama** [reply]
Ah, understand. So let‘s take the version numbers we see in Terminal..
Maybe, we should increase version to 1.0.0 or so?

**Reply by MadOne** [reply]
The integration state would allow a 1.x.x. we made great Progress since 0.0.7. This is no alpha/beta anymore.  I will try if it fixes the error. This are things required for official home assistant integrations and so no requirement fir us. But if they require it must have some advanteges. 

Further there is a integration quality scale:
https://developers.home-assistant.io/docs/integration_quality_scale_index/
We should go through that list and check if we fulfill that requirements and if we should.

**Reply by MadOne** [reply]
I pinned the versions of the requirements.
Version is not allowed in core integrations, but mandatory in custom integrations. So we can ignore that warning.
Changed the version to 1.0.0pre1

**Reply by MadOne** [reply]
We should release 1.0.0 before integrateing ito HACS.

**Reply by OStrama** [reply]
Do you build the release?

**Reply by MadOne** [reply]
We should build an rc and test it. The upgrade paths and so in. I thought about mirroring this to my original integration to force update anyone still uding my integration.

**Reply by OStrama** [reply]
Yes, good idea!

**Reply by OStrama** [reply]
Is it possible to state diffewrent version numbers? Or is a higher number also allowed? 
I recently upgraded to HA 2024.10 and now I have pymodbus 3.6.9 installed.
(I'd like top avoid that we have to update the manifest with every new HA version..)

**Reply by MadOne** [reply]
Maybe we should just go back to unpinned versions as we are not a native ha integration.

**Reply by OStrama** [reply]
Let's try with your settings in Manifest first and look what happens. Maybe HA is intelligent enough to handle it..

**Reply by OStrama** [reply]
Made a try: When pinning with "==" I see an error in my HA version 2024.10. Whern I use ">=" it works. Will it pass the test?

**Comment by OStrama** [comment]
Yes, makes sense. I will be available Mon Tue to adaptations regardin hacs integration.
 But doing 1.0.0 before absolutely makes sense.

**Comment by OStrama** [comment]
<img width="414" alt="image" src="https://github.com/user-attachments/assets/56f18513-a419-4218-a716-4a2d938dd7fc">

First time below zero. I think they use a special format for that (Einerkomplement?)

**Comment by MadOne** [comment]
Can't confirm at the moment. We had +2°C

**Reply by OStrama** [reply]
OK, here it was -0.5. 

If value > 32768 we have to substract 65536 then it fits with WEM portal.. And 32768 seems to be zero..
Will check it in later

**Comment by OStrama** [comment]
I check in the following changes:

- Temperature calculation as described above
- Changing Addr. 3311 to  "Vorlauftemperatur präzise"
- Updating config schema to version 4 and updating async_migrate_entry (When adding/changing config flow parameters we always should take care that default values are set during update..)

Would it make sense to build a 1.0.0pre for testing?

**Reply by MadOne** [reply]
Point 1 and 3 sounds good. What is Vorlauftemperatur präzise?

Jeah we should to some pre while still developing. Rc should be end of new features and just bugfixing.

**Reply by OStrama** [reply]
It's Vorlauftemperatur with a precision better than 0.5 degree ;-):
![image](https://github.com/user-attachments/assets/073b3df5-d55f-4869-bd08-5da85c0d55d4)

**Reply by MadOne** [reply]
So shouldent we drop the less precise value completely?

**Reply by OStrama** [reply]
I'm not sure if 33111 is available on every model. The info about 33111 is from here: https://github.com/OStrama/weishaupt_modbus/issues/20

**Reply by OStrama** [reply]
> Point 1 and 3 sounds good. What is Vorlauftemperatur präzise?
> 
> Jeah we should to some pre while still developing. Rc should be end of new features and just bugfixing.

OK, should we freeze features and shift the last 3 enhancements to 1.1.0?

**Reply by MadOne** [reply]
Yes i think so. We should make a new developement branch for new features and and the main one for bugfixes only.

**Reply by OStrama** [reply]
OK, makes sense..

**Reply by OStrama** [reply]
Cretated new branch Version-1.0 and made 1.0.0rc1

**Comment by OStrama** [comment]
I made some bugfixing in branch Version-1.0. 
Maybe, using the HA const for minutes in WW Push affects the translation you made?
I did not add translation up to now nor any changes in registers. Only missing spaces and crash if "Luftansaugtemperatur" not available. You were right, that it chrashed because of it.

**Reply by MadOne** [reply]
The translation system is quite a heavy change.  When its done we should merge all branches.

**Reply by OStrama** [reply]
Ahh, yes, sorry, was too fast with the debugging changes to main. Should I reverse?

**Reply by MadOne** [reply]
Nope its ok.

**Comment by OStrama** [comment]
@MadOne Made a pull request in your old repository by mistake. Please ignore and/or decline it..

**Comment by MadOne** [comment]
@OStrama : Somehow my dev container broke and i rebuild it. Since then no branch of our integration works. Do you have similar problems?

**Comment by OStrama** [comment]
No, I currently check translate branch. It works in general..

**Comment by MadOne** [comment]
```
  File "/workspaces/HA_core/config/custom_components/weishaupt_modbus/modbusobject.py", line 156, in value
    if len(mbr.registers) > 0:
           ^^^^^^^^^^^^^
AttributeError: 'ExceptionResponse' object has no attribute 'registers'
```

Even in the main branch. Maybe this is due to an homassistent update or the increase of pymobus?

**Comment by OStrama** [comment]
I switched from German to English and vice versa. Works well and entities remain. But you have to restart HA to change language. But this is OK I guess..

**Comment by MadOne** [comment]
Are sensor values and select values translated?

**Reply by OStrama** [reply]
Yes

**Reply by MadOne** [reply]
Weird it did not work for me. Well now nothing works...
Any chance you could join discord?
https://discord.gg/ZYUqBF8n
This would make conversation much easier....

**Comment by MadOne** [comment]
What pymobus version do you have? I have 3.7.4 now

**Reply by OStrama** [reply]
pymodbus in /home/vscode/.local/ha-venv/lib/python3.12/site-packages (3.6.8)

WHat kind of update did you do?

**Reply by OStrama** [reply]
and on my production system: pymodbus in ./lib/python3.12/site-packages (3.6.9)

**Reply by OStrama** [reply]
In the devcontainer: 
Successfully installed pymodbus-3.7.4

and a lot of exceptions:

  File "/workspaces/core/config/custom_components/weishaupt_modbus/modbusobject.py", line 156, in value
    if len(mbr.registers) > 0:
           ^^^^^^^^^^^^^
AttributeError: 'ExceptionResponse' object has no attribute 'registers'
